components/MainTiles/MainTiles.tsx

```diff
@@ -1,22 +1,27 @@
+import { useRouter } from 'next/dist/client/router';
 import Link from 'next/link';
 import type { ChangeEvent } from 'react';
-import { useCallback, useState, memo } from 'react';
+import { useCallback, memo } from 'react';
 
-import type { HomePageProps } from '../../pages';
-import { ArticleTile } from '..//ArticleTile/ArticleTile';
+import type { HomePageProps } from '../../pages/[displayStyle]';
 import { Button } from '../Button/Button';
 import { DisplayStyleSwitch } from '../DisplayStyleSwitch/DisplayStyleSwitch';
 
+import { BlogsGrid } from './BlogsGrid';
+import { BlogsList } from './BlogsList';
 import styles from './mainTiles.module.scss';
 
 type MainTilesProps = HomePageProps;
 
-export const MainTiles = memo<MainTilesProps>(({ blogs }) => {
-  const [displayStyle, setDisplayStyle] = useState<'GRID' | 'LIST'>('GRID');
+export const MainTiles = memo<MainTilesProps>((props) => {
+  const router = useRouter();
   const changeDisplayStyle = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => setDisplayStyle(e.target.value as 'GRID' | 'LIST'),
-    [],
+    (e: ChangeEvent<HTMLInputElement>) => {
+      return router.push(`/${e.currentTarget.value}`, undefined, { shallow: false });
+    },
```

Comment:
W tym przypadku `router.replace` może być lepsze, nie dodaje to nowego wpisu do historii użytkownika.

Reply:
Ale my raczej chcemy nowy wpis, nie?

Reply:
Jeśli zrobimy z `router.push` i po cichu zmienimy użytkownikowi url, to imo może być to dla niego zaskoczeniem, że `powrót do poprzedniej strony` tylko zmieni wygląd wyświetlania.

Reply:
Tak, również tak uważam. Robienie osobnych wpisów w historii tylko dla zmiany sposobu wyświetlania może wkurzyć. Zwłaszcza że button do przełączania raczej jest w tak intuicyjnym miejscu, że raczej nikt korzystając z przycisku do powrotu do poprzedniej strony, nie będzie chcieć po prostu zmienić tego widoku ;)